### PR TITLE
WRR-3609: Update major versions for dependencies

### PR DESCRIPTION
**Pull request by alexandrumorariu**
Enact-DCO-1.0-Signed-off-by: Alexandru Morariu alexandru.morariu@lgepartner.com

### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
I updated the following dependencies:
* readdirp from ^3.6.0 to ^4.0.1 (https://github.com/paulmillr/readdirp/releases/tag/4.0.0)
 Has breaking change for our project.
* query-string from ^8.2.0 to ^9.1.0 (https://github.com/sindresorhus/query-string/releases/tag/v9.0.0)
 No breaking change for our project.
* web-vitals removed from our project because it was not used inside the sample.

* @types/node from ^20.14.9 to ^22.5.2 (no revelvant README.MD or CHANGELOG.MD, I tested this manually)
 No breaking change for our project.

I also updated the scripts to work with the changes made in readdirp 4.0.1

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
In order to make our scrips work with the new version of readdirp I made some changes.
First readdirp.promise was replaced by readdirpPromise.
Second the import statement was changed from 
"const readdirp = require('readdirp')" to "const {readdirpPromise} = require('readdirp')".
Third the directoryFilter type was changed from an array of strings to a function. 
ex: from directoryFilter: ['!node_modules'] to directoryFilter: (di) => di.basename !== 'node_modules'.
Fourth the fileFilter type was changed from a string to a function. 
ex: from fileFilter: 'package.json' to fileFilter : (fi) => fi.basename === 'package.json'.
All the changes were inspired from their README.MD file (https://github.com/paulmillr/readdirp?tab=readme-ov-file#readdirp-)
examples 3) and 4).

I removed web-vitals from the project. I do agree with Daniel's comment that it was added based on our template.
### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRR-3609

### CODE REVIEW

File: sandstone/pattern-virtualgridlist-incremental-load/package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
 		"prop-types": "^15.8.1",
 		"react": "^18.3.1",
 		"react-dom": "^18.3.1",
-		"web-vitals": "^3.5.0"
+		"web-vitals": "^4.2.3"
```
Comment (daniel-stoian-lgp) [inline sandstone/pattern-virtualgridlist-incremental-load/package.json]: I see this dependency is not used in the sample.
The sample was added in #225 , probably based on the sandstone template, but the usage of web-vitals in the sample was removed.
Can you remove this dependency?

Comment (alexandrumorariu) [reply sandstone/pattern-virtualgridlist-incremental-load/package.json]: Done

**Review by daniel-stoian-lgp — APPROVED** [review]

File: scripts/bootstrap.js [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-	readdirp = require('readdirp'),
+	{readdirpPromise} = require('readdirp'),
 	shell = require('shelljs'),
 	path = require('path');
 
 let error = false;
 
-readdirp
-	.promise('.', {depth: 2, fileFilter: 'package.json', directoryFilter: ['!node_modules']})
+readdirpPromise('.', {depth: 2, directoryFilter: (di) => di.basename !== 'node_modules', fileFilter: 'package.json'})
```
Comment (hong6316) [inline scripts/bootstrap.js]: Could you please explain in more detail?
There is no migration guide for this change.
(ex: `directoryFilter` parameter was changed from `['!node_modules']` to `(di) => di.basename !== 'node_modules'`)

Comment (alexandrumorariu) [reply scripts/bootstrap.js]: This change is not very well documented from their side. After I updated the version all the scripts started failing. After an investigation I found out that the scripts only ran in the root repo and they never recursed until the depth 2, which is given as a parameter. I started removing one prop at a time to see what the problem was. When I removed "directoryFilter" the scripts ran OK, but they searched inside node_modules. Then I tried to change "directoryFilter" to work properly. The only clue I got that this change was necessary was their example ("directoryFilter: (d) => d.basename !== '.git'," from thier EXAMPLE section https://github.com/paulmillr/readdirp?tab=readme-ov-file#readdirp-). After that when you read their README.MD you find "directoryFilter: _**filter**_ to include/exclude directories found and to recurse into. Directories that do not pass a filter will not be recursed into." (from their API/options section https://github.com/paulmillr/readdirp?tab=readme-ov-file#readdirp-). From this and the example I took that we need to change the type of "directoryFilter" from an array of strings to a filter function. This function receives the name of the directory and returns true if you want to recurse into or false if you want to skip it.

Comment (alexandrumorariu) [reply scripts/bootstrap.js]: Even though the scripts work fine without this change, after reading their README.MD again I changed the fileFilter in the same way as directoryFilter. It seems the intended way to use this prop is also a function 
"fileFilter: filter to include or exclude files

    Function: a function that takes an entry info as a parameter and returns true to include or false to exclude the entry

" from https://github.com/paulmillr/readdirp?tab=readme-ov-file#options

**Review by hong6316 — APPROVED** [review]
👍

**Pull request merged**